evap/static/ts/src/infobox.ts

```diff
@@ -0,0 +1,40 @@
+declare const bootstrap: typeof import("bootstrap");
+
+import { selectOrError } from "./utils.js";
+
+const OPEN_CLOSE_TIMEOUT = 2000;
+
+export class InfoboxLogic {
+    private readonly infobox: HTMLDivElement;
+    private readonly closeButton: HTMLButtonElement;
+    private readonly storageKey: string;
+
+    constructor(infobox_id: string) {
+        this.infobox = selectOrError("#" + infobox_id);
+        this.closeButton = selectOrError("#" + infobox_id + " .callout-infobox-close");
+        this.storageKey = "infobox_" + infobox_id;
+    }
+
+    public attach = (): void => {
+        // close the infobox and save state
+        this.closeButton.addEventListener("click", async event => {
+            this.infobox.classList.add("closing");
+            setTimeout(() => {
+                this.infobox.classList.replace("closing", "closed");
+            }, OPEN_CLOSE_TIMEOUT);
+            event.stopPropagation(); // prevent immediate reopening
```

Comment:
Why would it reopen? This is the only event listener that should be active on the close button, right?

(especially consindering that the event handler below does not call this, but instead checks whether the initial state is closed)

Reply:
I improved the comment